src-tauri/src/manager.rs

```diff
@@ -330,29 +330,42 @@ fn start_module_thread(name: String, custom_args: Option<Vec<String>>, tx: Sende
 #[cfg(unix)]
 fn get_modules_in_path() -> BTreeSet<String> {
     let excluded = ["awk", "aw-tauri", "aw-client", "aw-cli"];
-    env::var_os("PATH")
-        .map(|paths| {
-            env::split_paths(&paths)
-                .flat_map(|path| fs::read_dir(path).ok())
-                .flatten()
-                .filter_map(Result::ok)
-                .filter_map(|entry| {
-                    let metadata = entry.metadata().ok()?;
-                    let is_executable = (metadata.is_file() || metadata.is_symlink())
-                        && metadata.permissions().mode() & 0o111 != 0;
-                    if !is_executable {
-                        return None;
-                    }
-
-                    entry
-                        .file_name()
-                        .to_str()
-                        .map(|s| s.to_string())
-                        .filter(|name: &String| name.starts_with("aw") && !name.contains("."))
-                })
-                .collect::<BTreeSet<_>>()
+
+    // Get the discovery path from config
+    let config = crate::get_config();
+
+    // Get the current PATH
+    let path = env::var_os("PATH").unwrap_or_default();
+    let mut paths = env::split_paths(&path).collect::<Vec<_>>();
+
+    // Add discovery path if not already in PATH
+    if !paths.contains(&config.defaults.discovery_path) {
+        paths.push(config.defaults.discovery_path.to_owned());
```

Comment:
This should be pushed to the front of the list so that it has priority (following PATH-lookup behavior).